internal/clients/telegram/client.go

```diff
@@ -0,0 +1,282 @@
+package telegram
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"net/url"
+	"path"
+	"strconv"
+	"sync"
+	"time"
+)
+
+// A client to interact with the telegram API. Start the client and then call Stop to stop it gracefully.
+// The client may take some time to shutdown if it has work to do.
+//
+// # Example
```

Comment:
do you really need this example in comment?

Reply:
Yes, because it explains how to use the telegram client. Just creating one isn't enough as it has to started and stopped

Reply:
I will add more comments as a shorter version of this 
https://gist.github.com/m-kuzmin/f6675dad25fc74daacef3c7d0b5d2375

Reply:
https://github.com/m-kuzmin/daily-reporter/pull/2/commits/0ad959982399b5b4623e57ce30b55c18c4851fb4